File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/GameRolePlayNpcInformations.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameRolePlayNpcInformations extends GameRolePlayActorInformations {
  override val typeId = 156

  protected trait Type extends GameRolePlayActorInformations.Type with Serializable {
    override val typeId = GameRolePlayNpcInformations.typeId

    val npcId: Short
    val sex: Boolean
    val specialArtworkId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(npcId)
      buf.writeBoolean(sex)
      buf.writeShort(specialArtworkId)
    }

  }

  protected def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, npcId: Short, sex: Boolean, specialArtworkId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val contextualId = buf.readInt()
    val look = EntityLook.deserialize(buf)
    val disposition = types.get[EntityDispositionInformations.TypeImpl](buf.readShort());
    val npcId = buf.readShort()
    val sex = buf.readBoolean()
    val specialArtworkId = buf.readShort()

    create(contextualId, look, disposition, npcId, sex, specialArtworkId)
  }

}

object GameRolePlayNpcInformations extends GameRolePlayNpcInformations {
  case class TypeImpl(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, npcId: Short, sex: Boolean, specialArtworkId: Short) extends Type
  override def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, npcId: Short, sex: Boolean, specialArtworkId: Short) = TypeImpl(contextualId, look, disposition, npcId, sex, specialArtworkId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, npcId: Short, sex: Boolean, specialArtworkId: Short) = create(contextualId, look, disposition, npcId, sex, specialArtworkId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameRolePlayNpcInformations.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameRolePlayNpcInformations extends GameRolePlayActorInformations {
    public static final short TYPE_ID = 156;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short npcId;
    public boolean sex;
    public short specialArtworkId;
    
    public GameRolePlayNpcInformations() { }
    
    public GameRolePlayNpcInformations(int contextualId, EntityLook look, EntityDispositionInformations disposition, short npcId, boolean sex, short specialArtworkId) {
        super(contextualId, look, disposition);
        this.npcId = npcId;
        this.sex = sex;
        this.specialArtworkId = specialArtworkId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(npcId);
        buf.writeBoolean(sex);
        buf.writeShort(specialArtworkId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        npcId = buf.readShort();
        if (npcId < 0)
            throw new RuntimeException("Forbidden value on npcId = " + npcId + ", it doesn't respect the following condition : npcId < 0");
        sex = buf.readBoolean();
        specialArtworkId = buf.readShort();
        if (specialArtworkId < 0)
            throw new RuntimeException("Forbidden value on specialArtworkId = " + specialArtworkId + ", it doesn't respect the following condition : specialArtworkId < 0");
    }
    
}
